src/panoramic/cli/analytics.py

```diff
@@ -0,0 +1,199 @@
+import json
+import logging
+import platform
+import shutil
+import time
+import uuid
+from datetime import datetime, timedelta
+from typing import Any, Dict, List
+
+# Unfortunate naming choice of Segment analytics python lib.
+# May be fixed in future, follow: https://github.com/segmentio/analytics-python/issues/154 .
+import analytics  # type: ignore
+
+from panoramic.cli.__version__ import __version__
+from panoramic.cli.config.analytics import get_write_key
+from panoramic.cli.config.analytics import is_enabled as config_is_enabled
+from panoramic.cli.config.storage import read_config, update_config
+from panoramic.cli.file_utils import (
+    append_json_line,
+    read_json_lines,
+    read_yaml,
+    truncate_file,
+    write_yaml,
+)
+from panoramic.cli.paths import Paths
+from panoramic.cli.print import echo_info
+
+logger = logging.getLogger(__name__)
+
+BLACKLISTED_ARGS = ['password']
+MINIMAL_FLUSH_DURATION = timedelta(minutes=10)
+MINIMAL_FLUSH_EVENTS = 10
+
+"""
+Usage: currently only way to track data is by using `write_command_event` after execution of each CLI command.
+If we decide to extend analytics with extra information, implement your own method like `write_request_event`
+that will track your data, but there is no need to call flush at the end.
+Data will be stored to events journal and flushed based on conditions set here.
+"""
+
+
+def write_command_event(name: str, group: str, params: Dict[str, Any], start_time: float, error: str = '') -> None:
+    """
+    Generate command execution context and store it in analytics events file as JSON line.
+    Truncate blacklisted parameters, like password.
+    Try to flush data to external source based on various conditions.
+    """
+    if not is_enabled():
+        return
+
+    for arg in BLACKLISTED_ARGS:
+        if arg in params:
+            del params[arg]
+
+    properties = {
+        'command': {
+            'name': ' '.join((group, name)),
+            'args': params,
```

Comment:
This seems like it would leak private information that we shouldn't be collecting. I realize we might blind ourselves a bit, but from an analytics perspective, the function name, user ID and runtime should be more than enough to figure out how people are using the CLI.

Reply:
Currently we have a blacklist of sensitive params, are you afraid that someone adds new command and won't know that he needs to fill some blacklist? Do we maybe want to switch to whitelist method or its not worth the effort and we can skip sending params altogether for now?

Reply:
I think we should stop sending params. They don't give us a ton of information that we need, and I worry the information that might leak would be problematic.

Reply:
ok, thats reasonable, lets try it without params, I will remove them